src/team_lib/mhba/xccl_mhba_mkeys.c

```diff
@@ -179,55 +161,123 @@ static xccl_status_t poll_umr_cq(xccl_mhba_context_t *ctx) {
 // Execute the UMR WQE for populating the UMR's MasterMKey
 static xccl_status_t
 populate_mkey(xccl_mhba_context_t *ctx, xccl_mhba_node_t *node, int mem_access_flags, struct mlx5dv_mkey *mkey,
-              void *mkey_entries) {
+              void *mkey_entries, int block_size, int team_size) {
     xccl_status_t status;
-    xccl_mhba_debug("Execute the UMR WQE for populating the MasterMKeys");
     ibv_wr_start(ctx->umr_qpx);
     ctx->umr_qpx->wr_id = 1; // First (and only) WR todo: check meaning
-    int repeat_count = node->sbgp->group_size / node->block_size; // todo check calc
-    mlx5dv_wr_mr_interleaved(ctx->umr_mlx5dv_qp_ex, mkey, mem_access_flags,
-                             repeat_count, node->sbgp->group_size,
-                             (struct mlx5dv_mr_interleaved *) mkey_entries); // todo check for error
+    if (block_size) {
```

Comment:
block_size = 0 has a special meaning which is unclear. lets have another flag that explicitly specifies the behavior